crypto/ec/ecp_secp384r1.c

```diff
@@ -0,0 +1,19594 @@
+ * SOFTWARE.
+ */
+
+/* Autogenerated: word_by_word_montgomery --static secp384r1 64 '2^384 - 2^128 - 2^96 + 2^32 - 1' */
+/* curve description: secp384r1 */
+/* machine_wordsize = 64 (from "64") */
+/* requested operations: (all) */
+/* m = 0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffeffffffff0000000000000000ffffffff (from "2^384 - 2^128 - 2^96 + 2^32 - 1") */
+/*                                                                    */
+/* NOTE: In addition to the bounds specified above each function, all */
+/*   functions synthesized for this Montgomery arithmetic require the */
+/*   input to be strictly less than the prime modulus (m), and also   */
+/*   require the input to be in the unique saturated representation.  */
+/*   All functions also ensure that these two properties are true of  */
+/*   return values.                                                   */
+/*  */
+/* Computed values: */
+/* eval z = z[0] + (z[1] << 64) + (z[2] << 128) + (z[3] << 192) + (z[4] << 256) + (z[5] << 0x140) */
+/* bytes_eval z = z[0] + (z[1] << 8) + (z[2] << 16) + (z[3] << 24) + (z[4] << 32) + (z[5] << 40) + (z[6] << 48) + (z[7] << 56) + (z[8] << 64) + (z[9] << 72) + (z[10] << 80) + (z[11] << 88) + (z[12] << 96) + (z[13] << 104) + (z[14] << 112) + (z[15] << 120) + (z[16] << 128) + (z[17] << 136) + (z[18] << 144) + (z[19] << 152) + (z[20] << 160) + (z[21] << 168) + (z[22] << 176) + (z[23] << 184) + (z[24] << 192) + (z[25] << 200) + (z[26] << 208) + (z[27] << 216) + (z[28] << 224) + (z[29] << 232) + (z[30] << 240) + (z[31] << 248) + (z[32] << 256) + (z[33] << 0x108) + (z[34] << 0x110) + (z[35] << 0x118) + (z[36] << 0x120) + (z[37] << 0x128) + (z[38] << 0x130) + (z[39] << 0x138) + (z[40] << 0x140) + (z[41] << 0x148) + (z[42] << 0x150) + (z[43] << 0x158) + (z[44] << 0x160) + (z[45] << 0x168) + (z[46] << 0x170) + (z[47] << 0x178) */
+
+#include <stdint.h>
+typedef unsigned char fiat_secp384r1_uint1;
+typedef signed char fiat_secp384r1_int1;
+typedef signed __int128 fiat_secp384r1_int128;
+typedef unsigned __int128 fiat_secp384r1_uint128;
+
+#if (-1 & 3) != 3
+#error "This code only works on a two's complement system"
+#endif
+
+/*
+ * The function fiat_secp384r1_addcarryx_u64 is an addition with carry.
+ * Postconditions:
+ *   out1 = (arg1 + arg2 + arg3) mod 2^64
+ *   out2 = ⌊(arg1 + arg2 + arg3) / 2^64⌋
+ *
+ * Input Bounds:
+ *   arg1: [0x0 ~> 0x1]
+ *   arg2: [0x0 ~> 0xffffffffffffffff]
+ *   arg3: [0x0 ~> 0xffffffffffffffff]
+ * Output Bounds:
+ *   out1: [0x0 ~> 0xffffffffffffffff]
+ *   out2: [0x0 ~> 0x1]
+ */
+static void fiat_secp384r1_addcarryx_u64(uint64_t *out1,
+                                         fiat_secp384r1_uint1 *out2,
+                                         fiat_secp384r1_uint1 arg1,
+                                         uint64_t arg2, uint64_t arg3) {
+    fiat_secp384r1_uint128 x1;
+    uint64_t x2;
+    fiat_secp384r1_uint1 x3;
+    x1 = ((arg1 + (fiat_secp384r1_uint128)arg2) + arg3);
```

Comment:
I'm not sure if I should look at the coding style, since this is auto generated code. But our coding style would have the `{` on the next line, and have an empty line before this one.

Reply:
I don't think we'll be updating fiat-crypto to allow parameterizing over coding styles.  We could add an extra empty line before this one if it's very important to you, but I'd like to leave the curly braces as-is.  So if you do want consistent style, even with the generated code, I'd suggest running the generated code through a code reformatter.

Reply:
@kroeckx 

> our coding style would have

One of the last ECCKiila [passes](https://gitlab.com/nisec/ecckiila/-/blob/master/ecp/@@ecp@@_CMakeLists.txt#L142) is `clang-format`. If there's something particularly egregious you see in the output, I'm happy to add it as an option there if you let me know what the option is -- I'm no `clang-format` expert. (E.g. @romen suggested the `ReflowComments: false` option.)